ms2query/ms2library.py

```diff
@@ -0,0 +1,270 @@
+from ms2query.query_from_sqlite_database import get_spectra_from_sqlite, \
+    get_tanimoto_score_for_inchikeys
+from typing import List, Dict, Union
+from matchms.Spectrum import Spectrum
+import pandas as pd
+import numpy as np
+import sqlite3
+import ast
+import pickle
+from gensim.models import Word2Vec
+from tqdm import tqdm
+from spec2vec import SpectrumDocument
+from spec2vec.vector_operations import calc_vector, cosine_similarity_matrix
+from ms2query.ms2query.query_from_sqlite_database import convert_array
+from ms2query.ms2query.s2v_functions import \
+    post_process_s2v
+
+
+class Ms2Library:
+    def __init__(self,
+                 sqlite_file_location: str,
+                 model_file_name: str,
+                 pickled_embeddings_file_name: str,
+                 **settings):
+        """
+
+        Args:
+        -------
+        sqlite_file_location:
+            The location at which the sqlite_file_is_stored. The file is
+            expected to have 3 tables: tanimoto_scores, inchikeys and
+            spectra_data. If no sqlite file is available paramater is expected
+            to be None.
+        file_name_dict:
+            A dictionary containing the files needed to create a sql database.
+            The dictionary is expected to contain the keys:
+            "npy_file_location", "pickled_file_name" and "csv_file_name". With
+            as value the location of these files.
+            Default = None
+        """
+        # Set default settings
+        self.mass_tolerance = 1.0
+
+        # Change default settings to settings given as kwargs
+        allowed_arguments = self.__dict__
+        for key in settings:
+            if key in allowed_arguments:
+                if isinstance(settings[key], allowed_arguments[key]):
+                    setattr(self, key, settings[key])
+                else:
+                    print(f"Different type is expected for argument: {key}")
+            else:
+                print(f"Invalid argument in constructor:{key}")
```

Comment:
```suggestion
            assert (key in allowed_arguments), f"Invalid argument in constructor:{key}"
            assert isinstance(settings[key], allowed_arguments[key]), f"Different type is expected for argument: {key}"
            setattr(self, key, settings[key])
```

Reply:
I would then also but this (starting at line 41 actually) into a separate class method, e.g. `def _get_default_settings()`.